### ISSUE TITLE
IntelliJ 2019.3 compatibility

### ISSUE THREAD

**Issue by krasa** [google/google-java-format#404]
You might want to use this: https://github.com/krasa/EclipseCodeFormatter/blob/master/src/java/krasa/formatter/plugin/ProjectCodeStyleInstaller.java#L86

The old way does no longer work:
```
com.intellij.diagnostic.PluginException: com.intellij.psi.codeStyle.CodeStyleManager requested as a service, but it is a component - convert it to a service or change call to project.getComponent()
	at com.intellij.ide.plugins.PluginManagerCore.createPluginException(PluginManagerCore.java:382)
	at com.intellij.diagnostic.PluginProblemReporterImpl.createPluginExceptionByClass(PluginProblemReporterImpl.java:12)
	at com.intellij.diagnostic.PluginException.createByClass(PluginException.java:58)
	at com.intellij.diagnostic.PluginException.logPluginError(PluginException.java:75)
	at com.intellij.serviceContainer.PlatformComponentManagerImpl.getService(PlatformComponentManagerImpl.kt:302)
	at com.intellij.openapi.components.ServiceManager.getService(ServiceManager.java:22)
	at com.intellij.psi.codeStyle.CodeStyleManager.getInstance(CodeStyleManager.java:40)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$null$0(ReformatCodeProcessor.java:116)
	at com.intellij.openapi.editor.ex.util.EditorScrollingPositionKeeper.perform(EditorScrollingPositionKeeper.java:100)
	at com.intellij.codeInsight.actions.ReformatCodeProcessor.lambda$prepareTask$1(ReformatCodeProcessor.java:106)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.openapi.application.impl.ApplicationImpl.runWriteAction(ApplicationImpl.java:885)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$preprocessFile$0(AbstractLayoutCodeProcessor.java:212)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$runProcessFile$2(AbstractLayoutCodeProcessor.java:309)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$null$5(AbstractLayoutCodeProcessor.java:392)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:220)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:178)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:168)
	at com.intellij.openapi.command.impl.CoreCommandProcessor.executeCommand(CoreCommandProcessor.java:154)
	at com.intellij.codeInsight.actions.AbstractLayoutCodeProcessor.lambda$null$6(AbstractLayoutCodeProcessor.java:390)
	at com.intellij.openapi.application.TransactionGuardImpl$2.run(TransactionGuardImpl.java:309)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.doRun(LaterInvocator.java:441)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.runNextEvent(LaterInvocator.java:424)
	at com.intellij.openapi.application.impl.LaterInvocator$FlushQueue.run(LaterInvocator.java:407)
	at java.desktop/java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:313)
	at java.desktop/java.awt.EventQueue.dispatchEventImpl(EventQueue.java:776)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:727)
	at java.desktop/java.awt.EventQueue$4.run(EventQueue.java:721)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:85)
	at java.desktop/java.awt.EventQueue.dispatchEvent(EventQueue.java:746)
	at com.intellij.ide.IdeEventQueue.defaultDispatchEvent(IdeEventQueue.java:906)
	at com.intellij.ide.IdeEventQueue._dispatchEvent(IdeEventQueue.java:779)
	at com.intellij.ide.IdeEventQueue.lambda$dispatchEvent$8(IdeEventQueue.java:422)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computePrioritized(CoreProgressManager.java:698)
	at com.intellij.ide.IdeEventQueue.dispatchEvent(IdeEventQueue.java:421)
	at java.desktop/java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:203)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:124)
	at java.desktop/java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:113)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:109)
	at java.desktop/java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.desktop/java.awt.EventDispatchThread.run(EventDispatchThread.java:90)
```

**Comment by plumpy** [comment]
Thanks!

Out of curiosity, does your code formatter plugin have a way of handling the imports too? I haven't really found a great way, so rely on people to configure their import settings (e.g. no wildcard imports) as a separate step, which is annoying...

**Comment by krasa** [comment]
Yes, I implemented my own import sorter -  IntelliJ's sorter is called first, and then I reshuffle imports using PSI (it must be done via PSI, or all kinds of side effects happen). 
But I did not bother with wildcard handling so `Class count to use import with` and `Name count to use static import with`  must be manually configured.
Also `Optimize imports on the fly` must be disabled - that would need a platform fix.

**Comment by plumpy** [comment]
Okay, I'll check out your code, thanks for the tip!

**Comment by krasa** [comment]
I think that you could just do this:

```
        <projectService serviceInterface="com.intellij.psi.codeStyle.CodeStyleManager" serviceImplementation="com.google.googlejavaformat.intellij.GoogleJavaFormatCodeStyleManager" overrides="true" />


public GoogleJavaFormatCodeStyleManager(@NotNull Project project) {
    super(new CodeStyleManagerImpl(project));
  }
```

I use a cglib proxy now, so I am stuck with programmatic registering.

**Comment by sormuras** [comment]
![image](https://user-images.githubusercontent.com/2319838/68279349-83299180-0073-11ea-9ae1-1f05bc3686bc.png)
Seems like this issue was fixed version `1.7.0.3`.

👍
